core/lib/zksync_core/src/lib.rs

```diff
@@ -630,6 +631,26 @@ pub async fn initialize_components(
                     Arc::new(ValidiumModeL1BatchCommitDataGenerator {})
                 }
             };
+
+        let current_commitment_mode_eth_response = eth_client
+            .call(
+                CallRequest::builder()
+                    .to((&contracts_config).executor_facet_addr.clone())
+                    .data(serde_json::from_str::<Bytes>(r#""0x252e5966""#).unwrap()) // The selector of "getPubdataPriceMode()"
+                    .build(),
+            )
+            .await
+            .unwrap();
+
+        let current_commitment_mode = L1BatchCommitDataGeneratorMode::from_eth_response(
+            &current_commitment_mode_eth_response,
+        )
+        .unwrap();
```

Comment:
Rather than unwrapping, I suggest we either report an error, pick a default, or directly `panic` in the implementation with a reasonable message.
Maybe include the actual read value in the error message.